### PR TITLE
Misc code updates

### PR DESCRIPTION
**Pull request by MSP-Greg**
Updates to code

### CODE REVIEW

File: src/testup/api.rb [hunk 1]
```diff
@@ -170,7 +170,7 @@ def self.suppress_warning_dialogs(&block)
         cache = ::Test.suppress_warnings?
         ::Test.suppress_warnings = true
       end
-      block.call
```
[the remark this answers is not in the document]
Comment (thomthom) [reply src/testup/api.rb:173]: I'm looking at this PR once commit at a time. Makes it easier to understand. (Large PRs aren't ideal)

So this relate to RuboCop performance suggestion?

I've always captured the block using `&block` when the method should always be called with a block. That way a call without a block throws an error. It's a little bit strange to see `yield` used when the block is captured. And doesn't this add an "unused variable warning"?

I'd like to keep the behaviour of errors being thrown to methods where blocks are not optional...

File: src/testup/file_reporter.rb [hunk 2]
```diff
@@ -124,7 +130,7 @@ def summary
   def log_basename
     # File system friendly version of ISO 8601. Makes the logs be sortable in
     # the file browser.
-    version = Sketchup.version.split('.').first
+    version = Sketchup.version[/\A\d+/]
```
Comment (thomthom) [inline src/testup/file_reporter.rb:133]: What's the reason for this change? Regexes isn't something everyone is comfortable with. Against I'd argue the original is clearer to read.

File: src/testup/gem_helper.rb [hunk 3]
```diff
@@ -6,62 +6,61 @@
...
       local_path = self.find_local_copy(gem_name, version)
       if local_path
         puts '> Installing from local copy...'
         puts "> #{local_path}"
         Gem.install(local_path, nil, {domain: :local})
         puts "> Installed #{gem_name}-#{version}"
       else
-        puts '> Installing from Ruby Gems...'
+        puts '> Installing from rubygems.org...'
+        version ||= Gem::Requirement.default
```
Comment (thomthom) [inline src/testup/gem_helper.rb:63]: Add a comment to what `version ||= Gem::Requirement.default` does/means?

File: src/testup/minitest_setup.rb [hunk 4]
```diff
@@ -18,73 +18,55 @@
...
-        result = self.testup_run_one_method(*args)
-        lapsed_time = Time.now - start_time
-        TestUp::Debugger.output("> Elapsed time: #{lapsed_time}s")
-        result
-      end
+module TestUp
+  module MinitestPrepend
+    def run_one_method(klass, method_name)
+      TestUp::Debugger.output("Running: #{klass.name}.#{method_name}")
+      start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
```
Comment (thomthom) [inline src/testup/minitest_setup.rb:28]: Can't we use the existing timer utility method here?

File: src/testup/minitest_setup.rb [hunk 5]
```diff
@@ -18,73 +18,55 @@
...
+        end
       end
-    end # class << self
-  end # class Runnable
+    end # def run
+  end
+end
+
+# TestUp modifications of Minitest
+Minitest.singleton_class.prepend TestUp::MinitestPrepend
```
Comment (thomthom) [inline src/testup/minitest_setup.rb:67]: Style: Use parentheses around arguments.

File: src/testup/debug_reporter.rb [hunk 6]
```diff
@@ -12,55 +12,6 @@
 require 'testup/minitest_setup.rb'
 require 'testup/app_files.rb'
 
-
-# Patching Minitest because we currently use 5.4.3 which doesn't have `prerecord`.
```
Comment (thomthom) [inline src/testup/debug_reporter.rb:16]: Good riddance to this!

**Review by thomthom — CHANGES_REQUESTED** [review]
Very nice changes Greg. Though I would have preferred several smaller PRs, as it's hard to review and requesting changes when there are multiple logical commits quickly makes it harder to manage t a clean git history.

File: src/testup/gem_helper.rb (hunk 3 above)
Comment (MSP-Greg) [reply src/testup/gem_helper.rb:63]: Do you mean explain `||=`?

`Gem::Version`, `Gem::Requirement`, and a string representation of a version (`"1.2.3"`) are all different.

File: src/testup/minitest_setup.rb (hunk 4 above)
Comment (MSP-Greg) [reply src/testup/minitest_setup.rb:28]: See above re time interval.

File: src/testup/file_reporter.rb (hunk 2 above)
Comment (MSP-Greg) [reply src/testup/file_reporter.rb:133]: This isn't a 'hot path', or, it's not called repeatedly.  Hence, speed isn't an issue.

You're implying that Ruby code published 'by Trimble' should be dumbed down for plugin authors.  I think it would be better to show concise, fast, and efficient code, regardless if it appears more complex.  Regex's aren't rocket science...

I'd prefer to leave the Regex, but I could change it to:
```
version, _ = Sketchup.version.split('.', 2)
```

File: src/testup/api.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (MSP-Greg) [reply src/testup/api.rb:173]: > I'm looking at this PR once commit at a time. Makes it easier to understand.

That's why I split it into several commits.  Your choice whether you squash them.

> So this relate to RuboCop performance suggestion?

Well, RuboCop often references the [`fast-ruby`](https://github.com/fastruby/fast-ruby) gem, and it has several helpful benchmark files. `yield` is faster, I tested with Ruby head.

> I've always captured the block using &block when the method should always be called with a block. That way a call without a block throws an error

So does the following if a block is not passed:
```ruby
def test(&block)
  yield
end
```

> And doesn't this add an "unused variable warning"?

No.  Block parameters are handled differently...

> I'd like to keep the behaviour of errors being thrown to methods where blocks are not optional...

See above.  Maybe check things like that before commenting.

File: src/testup/minitest_setup.rb (hunk 5 above)
Comment (MSP-Greg) [reply src/testup/minitest_setup.rb:67]: Passes RuboCop.  If people aren't sure what it does, the Ruby docs do not use parentheses...

**Comment by thomthom** [comment]
Besides some minor code style changes the majority of this PR has been merged as cherry picks in other PRs.

Thank you for all the fixes and improvements.
